### DISCUSSION TITLE
XCSoar on Kobo Clara Colour?

### DISCUSSION THREAD

**Discussion by ionelanton** [XCSoar/XCSoar#1433] [Ideas]
Hello,
Any intention to support this device?
Thank you!

**Comment by lordfolken** [comment]
Its a sweet target. If its built like the other kobos a port shouldnt be too difficult. Anyone un @XCSoar/kobo-team have such a device?

**Comment by fraktal-sb** [comment]
I could test a device but not buy it. I don't see the advantage of the colour, which is very transparent on e-ink colour screens

**Comment by ionelanton** [comment]
Hi, I bought a Kobo Clara Color. It does not have a SD card but a eMMC chip.
I tried to install XCSoar on it but without success. The device doesn't boot anymore. Bricked for now.

**Reply by fraktal-sb** [reply]
Have you used such an EMMC adapter for backup?

https://www.amazon.de/Hochgeschwindigkeits-EMMC-USB-3-0-Adapter-EMMC-Adapter-f%C3%BCr-EMMC-Modul-TF-Speicherkarten-Brennwerkzeug/dp/B0CRHQ356T

**Reply by lordfolken** [reply]
I believe the thing is soldered to the board.
https://guide-images.cdn.ifixit.com/igi/KcSJSHPMldedEWXk.huge

There is a "manual" reset procedure for this device, https://help.kobo.com/hc/en-us/articles/18332008301719-Manual-reset-Kobo-Clara-Colour-Kobo-Clara-BW

**Reply by Exadios** [reply]
Does anybody have a part number for the eMMC?

**Comment by fraktal-sb** [comment]
There seems to be a chance to do a backup/restore of the emmc chip

https://www.mobileread.com/forums/showthread.php?t=360628&page=2

**Reply by Exadios** [reply]
So the best thing would be to identify any connector / test pads corresponding to the JTAG if they exist and if the fuses have not been blown.

Otherwise, as somebody has pointed out the real problem is to 'dd' the saved image back onto the eMMC after it is (inevitably) bricked.

**Comment by Amilakvari** [comment]
Very interesting topic ! The color is the only missing thing on my Kobo Clara HD + Bluflyvario !!!!

**Comment by fraktal-sb** [comment]
the eMMC Chip should be this one, next to the test pad S1

![2024-05-31 17_52_52-KcSJSHPMldedEWXk huge (JPEG-Grafik, 1600 × 1200 Pixel) - Skaliert (76%) – Mozill](https://github.com/XCSoar/XCSoar/assets/69430537/5179df15-6614-4f64-aa5f-6d4cc8a4d36f)

**Reply by Exadios** [reply]
Yes. that is certainly it. Thanks.

**Reply by Exadios** [reply]
So it is a little difficult to tell from the photo. Is U22 mounted on some sort of sub board?

**Reply by fraktal-sb** [reply]
see link from lordfolken

https://guide-images.cdn.ifixit.com/igi/KcSJSHPMldedEWXk.huge

**Reply by Exadios** [reply]
Thanks. Not as I had hoped.

**Reply by ionelanton** [reply]
Good news! My device is now unbricked. It unbricked itself after I pushed the power button and let it alone with the white led blinking regularly and the screen being black. After a day I found the device showing the home screen. Connected it to my laptop and could see the drive's content. I tried to reset it from the screen menu and the screen froze showing a round loading icon. The device was in bricked state again. I let it like that a couple of days and today I pressed the power button while inserting the usb cable. The device started and requested a wifi connection, updated itself and after registering it again on Kobo site it is working normally.

**Reply by Exadios** [reply]
That is good, if confusing, news.

I think the best place to report this is on mobileread, maybe in [https://www.mobileread.com/forums/showthread.php?t=360628&page=2](this) thread. The reason that I say this is because, as I see it, step one is to get a dump of the disk as delivered from Kobo. This will invariably involve 'dd' and 'netcat'. So someone will have to craft a KoboRoot.tgz or Kobo.tgz to do this. I think some people on mobileread are also interested in this disk image, so better they do this rather than us. If not then we will have to bite the bullet.

Do you have an account on mobileread?

**Reply by ionelanton** [reply]
Hi, I do not have a mobileread.com account. I'll make one tomorrow. Thank you!

**Comment by lordfolken** [comment]
Did anyone solder a terminal to the TTY pads yet?

**Reply by Exadios** [reply]
Is there any chance of a shell there?

**Reply by lordfolken** [reply]
I would at least expect the boot messages to be visible on that.

**Comment by Pavekell** [comment]
Hello, any news on this topic ? Even with a manual installation or launch via NickelMenu and a script ? Thanks

**Reply by lordfolken** [reply]
You can install it. It will brick your device. The soldered chip means you cant undo the damage.

**Reply by Pavekell** [reply]
I tried installing it placing the latest KoboRoot.tgz file in the kobo connecting it to a computer but the only response I got was a black screen and a blinking light after kind of an update process on the kobo.
I had to manually reset the kobo via connecting to a power source, holding the button until no light, releasing the button, then holding it again until rapid blinking light and releasing it again when steady light.
I got back the factory reset and an operating kobo.

Then I tried via an other method:
Installing nickel menu with a specific menu item to launch a specific file in the kobo via a script.sh (unzipped binary file for xcsoar) but never worked.

do you think or ever tried this to have an operating XCsoar on KoboClara Color ?

Thanks

**Comment by fraktal-sb** [comment]
I don't see a chance:

https://www.mobileread.com/forums/showthread.php?t=363175

**Comment by yorickreum** [comment]
Somewhat related, there should be a fair chance of getting it to run on an Amazon Kindle... https://www.reddit.com/r/kindle/s/n0NpyrVW5v

**Comment by HudPesjan** [comment]
Just opened up a Clara BW and seem to have the same issue, no SD card.

**Comment by marcseiffert** [comment]
https://elixirforum.com/t/the-journey-to-running-nerves-on-a-kobo-clara-e-reader/73331
It seems thats the solution. Thanks to mlaiez.....

**Reply by lordfolken** [reply]
I'm a bit confused: It obviously uses fastboot, has some android kernel stuff, but its not an android os?
Fastboot is good news however, so it can be flashed like a standard android device. 

 But i want to make it clear; that with the current project load, i don't have time to do this. I will support anyone taking this up and help upstream the changes needed.

**Comment by Aussie-Hoss** [comment]
Does anybody have any idea on how to install this on the KOBO Clara Colour ?

**Reply by alexu-dev** [reply]
I've been intrigued enough to do some basic investigation on this. These answers are based on my current knowledge.

> lordfolken wrote:
> I'm a bit confused: It obviously uses fastboot, has some android kernel stuff, but its not an android os?

From the evidence provided, yes. It's not a typical setup, so if forced to guess why I would say Kobo started from a MediaTek reference design for an Android tablet, deleted most but not all of the the Android stuff and changed the display to E-ink.

> Aussie-Hoss wrote:
> Does anybody have any idea on how to install this on the KOBO Clara Colour ?

Going from mlainez's blog, my understanding is that basic steps are:
 1 - Download the buildroot tree provided [here](https://github.com/Spin42/buildroot-kobo).
 2 - Build it.
 3 - Put the image on the mass storage partition of your Kobo.
 4 - Download the new init file provided [here](https://github.com/Spin42/kobo-init) (ignore the Nerves file).
 5 - SSH into your kobo and replace the original init file with the new one (note that mlainez's blog suggested getting the IP address from you DHCP, but you can also find it on the Kobo under Settings->Device Information).
 6 - Reboot the Kobo, press the button at the right moment, and the system will boot with the alternate file system.
 
NOTE: While I have activated Fastboot and poked around with SSH, I have *not* made any changes to my Kobo. I've just been reading docs to get a better understanding.
 
So now you have a terminal on a buildroot configured system that you control, but presumeably you want XCsoar. Building XCsoar is based on makefiles, and buildroot understands how to make makefile projects, so all we need to do is write a package that describes where to get the XCsoar tarball and 'make target=KOBO'. Simple right?

Unfortunately, no.

From the XCsoar Developer manual section 'Compiling for Kobo E-book Readers'
 1 - The current script will build a toolchain, something that buildroot has already done (and the current XCsoar script is unlikely to build the correct toolchain).
 2 - The current script assumes a custom linux kernal can be built and installed. According to mlainez the Clara Color uses secure boot, so the bootload and kernal cannot be changed (but the kernal can load additional unsigned modules).
 3 - The current script will build the final image file, something that buildroot already does.
 
So as I see it, the first step to getting XCsoar running is someone familiar with the XCsoar codebase and build system has to create a build target for the Clara Color that either plays nice with buildroot or replicates the bits that buildroot is providing.

That job is well above my paygrade.

**Comment by anj1** [comment]
I have a work in progress up and running, see https://github.com/XCSoar/XCSoar/issues/1707#issuecomment-4598108405

**Reply by lordfolken** [reply]
Way to go. :+1: 

What are the challenges for the download manager? DNS? 
I guess that hardware has no onboard gps like the other kobos.

**Reply by anj1** [reply]
No onboard gps but ample space for a module+antenna, and a UART as well.

**Reply by anj1** [reply]
I've pushed the required Nickel cross-compilation toolchain as well: ghcr.io/anj1/nickeltc-gcc10:latest

**Comment by marcseiffert** [comment]
HiKobo Hardware had never onboard gps. For this reason we use blueflyvario hardware solution or you bind with bluetooth or wlan to many varios with gps…..Von meinem iPhone gesendetAm 02.06.2026 um 08:39 schrieb A Nejati ***@***.***>:﻿
No onboard gps but ample space for a module+antenna, and a UART as well.

**Comment by anj1** [comment]
Update: I've gotten it working with a custom gps module which I put inside the case and connected to the tty. XCSoar boots up and runs normally. See my branch: https://github.com/anj1/XCSoar/tree/kobo-clara-colour. It's proving to be really good, the color helps more than I'd expected. And this is with the default color scheme; optimizing for more Kaleido 3-friendly color schemes will probably make it work a lot better.

<img width="1008" height="1481" alt="IMG_8696sm" src="https://github.com/user-attachments/assets/47c38b71-a122-4ee6-9945-5a4b9ea1b0f4" />

I had to do some modifications to various hardware configs and rendering.

I'll soon add a full how-to guide including how to put in a gps module.

@lordfolken @fraktal-sb  how do you suggest proceeding with integrating it into the main repo? You may want to have a look at the code.

**Reply by lordfolken** [reply]
Congratulations. :-) 

Please create PR after cleaning up your git history.  No fixups, and small contained changes that each should compile. 
Right now i see quite a bit of extra stuff in your git history that isn't kobo related. Also a couple of changes we would have to talk about..

**Reply by anj1** [reply]
I'm in the process of cleanup now. For not-Kobo-related stuff, I assume you mean e.g. this commit: https://github.com/XCSoar/XCSoar/commit/1ecd34d026363b5cd7a9716772c4d9315f823e53 

With these changes:
- 32-bit-safe UTC date conversion
- `PackedFloat` fallback without `constexpr` bit-cast
- `VersionNumber` `major`/`minor` macro cleanup
- `Path` formatter null handling

touching these files:
[src/lib/fmt/PathFormatter.hpp](https://github.com/XCSoar/XCSoar/commit/1ecd34d026363b5cd7a9716772c4d9315f823e53#diff-5a7d9a017a12d5e6dd03d6c093a102add9dcead156b9a67331404f8a272ff04f)
[src/time/BrokenDateTime.cpp](https://github.com/XCSoar/XCSoar/commit/1ecd34d026363b5cd7a9716772c4d9315f823e53#diff-348254daa39b37a9199ac91d46f30cdac61d80ce41a7e5e3939637dba65fd781)
**[src/util/PackedFloat.hxx](https://github.com/XCSoar/XCSoar/commit/1ecd34d026363b5cd7a9716772c4d9315f823e53#diff-348254daa39b37a9199ac91d46f30cdac61d80ce41a7e5e3939637dba65fd781)**
[src/util/VersionNumber.hxx](https://github.com/XCSoar/XCSoar/commit/1ecd34d026363b5cd7a9716772c4d9315f823e53#diff-348254daa39b37a9199ac91d46f30cdac61d80ce41a7e5e3939637dba65fd781)

(because the kobo/nickel cross-compiler/libstdc++ combination lacks std::bit_cast and __builtin_bit_cast.)

I agree these probably need to be separate; what do you suggest?

**Reply by lordfolken** [reply]
If its a dependency for the target, i'm fine to have it in the pr.

**Comment by fraktal-sb** [comment]
sorry for late reply...

@anj1 congratulations and "chapeau"also from my side. I realy appreciate your work. Lots of pilots waited for this solution.

My requirements are:

- the installation of the colour version should be the same as it is now (put the KoboRoot.tgz on the .kobo directory and reboot)
- if possible there should be a backup procedure available to save/restore the original firmware.
- it should still be possible to make own adaptions of XCSoar and compile and link it on a local linux system.

**Reply by anj1** [reply]
Thank you!

The install is via the Nickel toolchain so there's no need to change firmware, it's just an extra app you add to the Kobo. The original functionality still works (books etc), you just add the XCSoar app and launch via NickelMenu.

And yes linux cross-compilation is available. Once I finish with the cleanup I'll add more comprehensive build/deploy instructions. As with any open source project, help is welcome and appreciated as well!

**Comment by fraktal-sb** [comment]
@anj1 XCSoar as extra app is the right way and fully flexible. I could help on testing and documentation. For existing KOBO's I have a screenshot program which reads /dev/fb0 and converts the framebuffer image to a .png image. If you need this...

**Comment by gratefulfrog** [comment]
Hi,
I am very interested in upgrading from a Kobo Nia to a Clara color.

Is there any progress on this? Any way to help or to test?  

Is there currently a working version for clara color?

Cheers,
Bob

**Reply by lordfolken** [reply]
None from this project so far. Im kinda still waiting for @anj1 pull request.

**Comment by anj1** [comment]
I've been pulled away by other projects but have had time now to sit down and clean this up.

There's 3 parts: The XCSoar build itself, the supporting toolchain, and the script for actually running it on the kobo.

I've put my updated nickel toolchain here:
https://github.com/anj1/NickelTC/pkgs/container/nickeltc-gcc10

@lordfolken Instead of one overwhelmingly large PR, I'll first create a new discussion thread for the various changes that were required.

**Comment by gratefulfrog** [comment]
@anj1 
Thanks so much for your post and for all the work! Well done!

I'm very much interested in your work. See my post #2821.

I'm not sure I followed your information. Do you have a working method of making xcsoar run on a Kobo Clara Color?  Would you have any pictures or even better a video of it working? Even some description, i.e. are the colors ok, better than black and white? How about the touch response and general interaction? And reliability? Have you taken it out flying for a long flight? Did it crash?

I hesitate to buy another kobo, given then generally poor performance in comparison to an Android phone and/or Tablet. Any thoughts?

Again, thanks,
Bob

**Reply by anj1** [reply]
Yes there is a working method of making XCSoar run on the Kobo Clara Colour. It runs as a normal app. There are pictures further up in this thread: https://github.com/XCSoar/XCSoar/discussions/1433#discussioncomment-17527821

The colors work surprisingly well, better than I thought, but for the best experience we might need some tweaking of the map drawing styles and so on. Touch response works fine. It seems stable and I haven't experienced a crash yet. The Kobo Clara Colour uses a new MediaTek MT8113 chipset (very different from their prior chipsets) and it seems to handle the load without issue.

Hardware modification isn't required to run XCSoar, but what I've done on my own device is add a gps module inside the kobo case. I'm writing up a doc on how to do the hardware modification which I'll put up. The only issue I've experienced so far is that because the kobo hasn't really been designed to house an antenna, the antenna placement is a bit tricky, and gnss reception isn't as great as it could be. But once it's obtained a fix it seems stable.

**Reply by gratefulfrog** [reply]
That's fantastic!  You have really done some great hacking! Thank you so much!

I assume you soldered the GPS to the internal serial port?  So I could do the same with a bluefly TTL Vario/GPS, like I've done for so many Nias?

Or would a Bluetooth connection be more practical?

Do you have any thoughts about the pros/cons of the Kobo CLara Color vs. the Bigme B6 on Android 14?

Really, thank you so much!

**Reply by anj1** [reply]
Yes it's soldered to the 'J5' port, which is a serial port. You can solder any module that supports 1.8V uart serial communication (just as long as it fits inside the case -- or you could figure out a way to mount it outside the case). I haven't used the Bigme B6. Both it and the Kobo Clara Colour use the Kaleido 3 display. A great thing about the Kobo is that it's fairly easy to take the case off and it has a lot of spare internal space for putting in a gps module. I haven't seen any teardown photos of the B6, so if it doesn't have space for gps, options are limited.

**Reply by anj1** [reply]
@gratefulfrog description of hardware modifications here: https://anj1.github.io/research/posts/xcsoar-kobo-clara-colour/

**Comment by gratefulfrog** [comment]
Thanks again!

I mount my bluefly's in a 3d printed case on the outside of the Kobo.

For the Bigme, I would use a Bluetooth Vario/gps not a wired connection.

I'm thinking about the convenience and performance of a modern Android
device ....

**Reply by anj1** [reply]
The Kobo runs linux as well. Both run various semi-proprietary flavors of linux. The MT8113 in the kobo is very low power so battery life is excellent. XCSoar is not very processing heavy so the more power of the Helio P35 in the bigme doesn't offer a huge advantage. The Kobo OS is stripped down whereas full Android is quite processing heavy. One real advantage of the B6 is lack of secure boot.

**Reply by ulipo** [reply]
I think I'll soon gift myself a Clara Colour as my next instrument, when I will have it in my hands I'll surely design a case to be 3D printed to house a bluefly vario, [as I did with the GloHD and some others.](https://www.thingiverse.com/ulipo/collections/28344977/things).

**Reply by gratefulfrog** [reply]
@ulipo Could you share your  motivations for a Kobo rather than a Bigme B6 running android?  Personally, I think the Bigme solution is looking much more appealing than the idea of hacking a kobo and soldering a GPS/Vario inside it after printing a case...

**Reply by ulipo** [reply]
Well... hacking, soldering, 3d printing, that's more fun, no? ;-)
You could also buy a FlyMaster, a Digifly or whatever commercial instrument fits your cockpit, but you come here for XCSoar :-D

Also I don't like what google is doing with android, I would love to get rid of it even in my phone...

**Reply by gratefulfrog** [reply]
Thanks for the reply. I am just curious as to why pilots chose one option or another. I was not trying to convince you and certainly not offend you!

I fully agree with you about Google's overlord control of Android not being in my best interests.  

I also agree that hacking, printing, etc. is fun. I spend all my days and nights doing just that ...  (my current project is a full set of customizable RAM-like support mounts printed in ASA)

But fun does not alleviate the frustration of flying and having the Kobo crash after 2 hours, losing the barometric altitude and screwing up the recording... Nor the poor touch reactivity, or the poor map display and general slowness.

Most of the people in my club fly with xcsoar on a phone, except for those who own a glider and install a pro system on the control panel.  The general consensus is that having a good display and touch reactivity and all the advantages of a connected device outweigh the the downsides like overheating and sometimes being invisible because of the sun.

If I could afford a glider, I could afford to have a professional navigation system... but I cannot.

So I am just wondering about my next step : 
* Can I get my Kobo Nia to run xcsoar for a full flight without crashing?
* Is there an acceptable color screen alternative, e.g. Kobo Clara Color, or Bigme or other, that wouldn't crash, and wouldn't break the bank?
* Are the connectivity advantages of an Android device worth the compromises?

I don't have the answers to these questions, which is why I'm interested in pilot's experiences and opinions.

Thank you for helping!!!

**Reply by lordfolken** [reply]
> Can I get my Kobo Nia to run xcsoar for a full flight without crashing?
This shouldn't be happening and should be a bug report. Logs, version, profile would be interesting. 

The kobo clara color is a lot faster than the nia. Its got a dual core 2GHz CPU, compared to the single 900Mhz of the nia.

**Reply by anj1** [reply]
@gratefulfrog In addition to what lordfolken said, you can use the blueflyvario with the kobo clara colour as well; soldering an internal gps was simply my preference.

Could you explain the advantages you believe exist with a device like the B6? And I'd be grateful if you could do that in a new discussion thread, not this one.

**Reply by ulipo** [reply]
> Thanks for the reply. I am just curious as to why pilots chose one option or another. I was not trying to convince you and certainly not offend you!
> 
> I fully agree with you about Google's overlord control of Android not being in my best interests.
> 
> I also agree that hacking, printing, etc. is fun. I spend all my days and nights doing just that ... (my current project is a full set of customizable RAM-like support mounts printed in ASA)
> 
> But fun does not alleviate the frustration of flying and having the Kobo crash after 2 hours, losing the barometric altitude and screwing up the recording... Nor the poor touch reactivity, or the poor map display and general slowness.
> 
> Most of the people in my club fly with xcsoar on a phone, except for those who own a glider and install a pro system on the control panel. The general consensus is that having a good display and touch reactivity and all the advantages of a connected device outweigh the the downsides like overheating and sometimes being invisible because of the sun.
> 
> If I could afford a glider, I could afford to have a professional navigation system... but I cannot.
> 
> So I am just wondering about my next step :
> 
>     * Can I get my Kobo Nia to run xcsoar for a full flight without crashing?
> 
>     * Is there an acceptable color screen alternative, e.g. Kobo Clara Color, or Bigme or other, that wouldn't crash, and wouldn't break the bank?
> 
>     * Are the connectivity advanta

No offense taken and no offense intended!

I started with a Kobo Mini and a GPS chip, after that I used a Glo and then a GloHd with BlueFly. The Mini was a little bit sluggish but the Glo and GloHD never crashed one time.
I know the NIA is kinda a cheap device, probably Rakuten saved money on the processor...

Regarding the responsiveness of the touch-screen, flying with a paraglider means that i seldom touch it at all during the flight, actually I lock the screen before taking off and unlock it after landing.
Nonetheless, with my current GloHD it is far less responsive than a modern smartphone but doesn't feel laggy at all.

**Reply by gratefulfrog** [reply]
@anj1 
> Could you explain the advantages you believe exist with a device like the B6? And I'd be grateful if you could do that in a new discussion thread, not this one.  

I did try to start a new thread  #2821  but there wasn't much interest there.   I will reply to you there, though. I don't know how or even if it's possible to transfer this talk  to that thread? It seems that it's all happening at your end of town ;-)  

I'm sorry if this thread has been "hijacked" by the discussion that your work has provoked.  I am sincerely thankful to you and would be more than happy to be an alpha or beta tester of your stuff on a Clara Color. Just give me some warning so I can get my hands on one...

**Comment by gratefulfrog** [comment]
Thanks for that info!

I was thinking of all the convenient things that a tablet could do, like
run a web browser or apps to display the igc files, or communicate over
internet etc.

The Kobo Nia that I use is only good as pilot support, nothing else. And
even that is pretty mediocre when compared to xcsoar on a modern phone.

I worry that using a Clara, I will only get a marginally better map display
and nothing more. And that it may be hard to use and/or maintain.

I wish I had one to try, and a bigme as well, but I can't spend so much
just to experiment...

**Comment by marcseiffert** [comment]
Get High Stay Highgethighstayhigh.blueflyvario.comHere you find many useful infos how to for different Kobo models.I use for more than 10 years Kobo ereaders as flight instrument. Am 10.08.2026 um 19:47 schrieb gratefulfrog ***@***.***>:﻿
Thanks for the reply. I am just curious as to why pilots chose one option or another. I was not trying to convince you and certainly not offend you!
I fully agree with you about Google's overlord control of Android not being in my best interests.
I also agree that hacking, printing, etc. is fun. I spend all my days and nights doing just that ...  (my current project is a full set of customizable RAM-like support mounts printed in ASA)
But fun does not alleviate the frustration of flying and having the Kobo crash after 2 hours, losing the barometric altitude and screwing up the recording... Nor the poor touch reactivity, or the poor map display and general slowness.
Most of the people in my club fly with xcsoar on a phone, except for those who own a glider and install a pro system on the control panel.  The general consensus is that having a good display and touch reactivity and all the advantages of a connected device outweigh the the downsides like overheating and sometimes being invisible because of the sun.
If I could afford a glider, I could afford to have a professional navigation system... but I cannot.
So I am just wondering about my next step :

Can I get my Kobo Nia to run xcsoar for a full flight without crashing?
Is there an acceptable color screen alternative, e.g. Kobo Clara Color, or Bigme or other, that wouldn't crash, and wouldn't break the bank?
Are the connectivity advantages of an Android device worth the compromises?

I don't have the answers to these questions, which is why I'm interested in pilot's experiences and opinions.
Thank you for helping!!!

**Comment by fraktal-sb** [comment]
@gratefulfrog Did you use a Wifi connection to a SoftRF device when you had "Kobo crash after 2 hours"?

Some years ago I found out by chance when omitting a parameter in src/Kobo/System.cpp XCSoar continues operation. I could share a personal version if you are interested. I have best experiences with KOBO GLO and GLO-HD as well with TOUCH. The XCSoar KOBO Nia support was never good (my impression). Most critical components in my configuration has been the Stratux Power supply cables. They have been the most source of GPS reception problems.

In general (my Opinion) an App for KOBO ist the best way for those who want XCSoar on a color e-ink device. Chapeau to anj1

**Reply by gratefulfrog** [reply]
@fraktal-sb 
No, My Kobo is 100% stand alone. I have soldered a bluefly TTL to the serial port and mounted it in a 3D printed enclosure.  I do not use an external battery. It works well for 2 to 3 hours, then crashes in flight, nearly every time. I am using version 7.44, and it was the same for the previous version as well. I have played around with the info boxes, and that seems to have gotten it to run longer. A friend in my club, using the same kobo+bluefly has similar crashes, but with a completely different profile config. 

The crashes are either a complete freeze or a spontaneous restart.

As I said with @lordfolken , when I replay the flight, the logs don't show anything...

I am happy to participate in debugging, but how?  Maybe on the next crash?

**Reply by twhitehead** [reply]
Did some [hacking on the Kobo Aura back in the day](https://github.com/twhitehead/notes-kobohacking). Have been using it ever since, and it has been rock solid for years of many extended flight (5+hrs).

I have some vague recollection that there was some sort of cross use thing going on with the serial port. Unfortunately I can't check it as I just sat on it last weekend and killed the top half of the screen (hence why I am lurking here considering getting a Clara Colour).

I have a Mini though, so I installed XCSoar on it and it looks like the kernel is set to run a console on ttymxc0
```
~ # cat /proc/cmdline 
console=ttymxc0,115200 rootwait rw quiet lpj=3997696 root=/dev/mmcblk0p1 rootfstype=ext4 hwcfg_p=0x7ffffe00 hwcfg_sz=512 waveform_p=0x7ff17800 waveform_sz=951808 mem=255M
```
I wouldn't necessary expect this to cause a hang, as, on the Kobo side you would have

  XCSoar - reading only (?)
  kernel - writing only (presumably no matching getty running?)

and on the bluefly side you would have

  bluefly - writing NEMA records
  bluefly - reading command (but hopefully not getting triggered by any kernel messages)

If you don't want to mess around with your uboot boatargs (there are direction on how to do this in my notes)
```
~ # dd if=/dev/mmcblk0 bs=1024 skip=768 count=128 | strings
...
bootcmd_mmc=run bootargs_base bootargs_mmc;load_ntxkernel; bootm
...
bootcmd=run bootcmd_mmc
...
bootargs_mmc=setenv bootargs ${bootargs} root=/dev/mmcblk0p1 rootfstype=ext4
...
bootargs_base=setenv bootargs console=ttymxc0,115200 rootwait rw quiet lpj=3997696
...
```
you could probably just add a line to your `/opt/xcsoar/bin/rcS` to disable printing anything but KERN_EMERG messages on the console
```
sysctl -w kernel.printk="0 4 1 7"
```

I know it doesn't entirely make sense that this would help with freezing, but my memory is that it might have. If it does, please let me know. I also ran a custom kernel on my Aura with my USB interupt fix, so that might have been required as well. Good luck.
